apps/entity-service/src/entities/file-upload.controller.ts

```diff
@@ -1,28 +1,71 @@
-import { Controller, Post, Param, NotFoundException, Req } from "@nestjs/common";
-import { FileUploadService } from "../file/file-upload.service";
+import {
+  Controller,
+  Post,
+  Param,
+  NotFoundException,
+  UnauthorizedException,
+  UseInterceptors,
+  UploadedFile,
+  Body,
+  BadRequestException,
+  InternalServerErrorException
+} from "@nestjs/common";
+import { ExtractedRequestData, FileUploadService } from "../file/file-upload.service";
 import { EntitiesService } from "./entities.service";
 import { EntityModel } from "@terramatch-microservices/database/constants/entities";
 import { PolicyService } from "@terramatch-microservices/common/policies/policy.service";
-import { FastifyRequest } from "fastify";
 import { MediaCollectionEntityDto } from "./dto/media-collection-entity.dto";
+import { ExceptionResponse } from "@terramatch-microservices/common/decorators/exception-response.decorator";
+import { ApiExtraModels, ApiOperation } from "@nestjs/swagger";
+import { FileInterceptor } from "@nestjs/platform-express";
+import "multer";
+import { buildJsonApi } from "@terramatch-microservices/common/util/json-api-builder";
+import { MediaDto } from "./dto/media.dto";
+import { MediaService } from "@terramatch-microservices/common/media/media.service";
 
 @Controller("entities/v3/files")
+@ApiExtraModels(MediaDto)
```

Comment:
I believe this won't be needed once the correct `@JsonApiResponse` decorator gets added to the endpoint.